web/src/app/lists/PaginatedList.tsx

```diff
@@ -52,136 +61,66 @@ const useStyles = makeStyles(theme => ({
   },
 }))
 
-function PageControls(props: {
-  isLoading: boolean
-  onNext?: () => void
-  onBack?: () => void
-}) {
-  const classes = useStyles()
-  const { isLoading, onBack, onNext } = props
-
-  return (
-    <Grid
-      item // item within main render grid
-      xs={12}
-      container // container for control items
-      spacing={1}
-      justify='flex-end'
-      alignItems='center'
-      className={classes.controls}
-    >
-      <Grid item>
-        <IconButton
-          title='back page'
-          data-cy='back-button'
-          disabled={!onBack}
-          onClick={() => {
-            onBack && onBack()
-            window.scrollTo(0, 0)
-          }}
-        >
-          <LeftIcon />
-        </IconButton>
-      </Grid>
-      <Grid item>
-        <IconButton
-          title='next page'
-          data-cy='next-button'
-          disabled={!onNext}
-          onClick={() => {
-            onNext && onNext()
-            window.scrollTo(0, 0)
-          }}
-        >
-          {isLoading && !onNext && (
-            <CircularProgress
-              color='secondary'
-              size={24}
-              className={classes.progress}
-            />
-          )}
-          <RightIcon />
-        </IconButton>
-      </Grid>
-    </Grid>
-  )
-}
-
-const loadingStyle = {
-  color: 'lightgrey',
-  background: 'lightgrey',
-  height: '10.3333px',
-}
-
-const useLoadingStyles = makeStyles({
-  item: {
-    display: 'block',
-    minHeight: dense => (dense ? 57 : 71),
-  },
-  lineOne: {
-    ...loadingStyle,
-    width: '50%',
-  },
-  lineTwo: {
-    ...loadingStyle,
-    width: '35%',
-    margin: '5px 0 5px 0',
-  },
-  lineThree: {
-    ...loadingStyle,
-    width: '65%',
-  },
-})
-
-// LoadingItem is used as a placeholder for loading content
-function LoadingItem(props: { dense?: boolean }) {
-  const classes = useLoadingStyles(props.dense)
-
-  return (
-    <ListItem className={classes.item} dense={props.dense}>
-      <ListItemText className={classes.lineOne} />
-      <ListItemText className={classes.lineTwo} />
-      <ListItemText className={classes.lineThree} />
-    </ListItem>
-  )
-}
-
 export interface PaginatedListProps {
-  headerNote?: ReactNode
+  // cardHeader will be displayed at the top of the card
+  cardHeader?: ReactNode
+
+  // listHeader will be displayed at the top of the list
+  listHeader?: ReactNode
 
-  items: PaginatedListItemProps[]
+  items: PaginatedListItemProps[] | ControlledPaginatedListItemProps[]
+  itemsPerPage?: number
 
   isLoading?: boolean
-  loadMore?: (numberToLoad: number) => void
+  loadMore?: (numberToLoad?: number) => void
```

Comment:
Do we actually support/use `numberToLoad`? Since this is the component triggering the event, I'd say we just remove the argument if we're never sending it.

Reply:
We are using it in PaginatedList to render twice the amount of items for the next page, but on infinite scrolling for instance we use the original itemsPerPage

Reply:
Based on `QueryList` current state, it looks like we ignore this value and use `queryVariables.input.first` instead.

Reply:
Ah, I’ll ensure we’re covering both use cases then

Reply:
Looks like that `once()` call is taking a callback from `loadMore`, similar to events passed to `onClicks` if you set by reference.

So we'll need to use that param for page lists, but use itemsPerPage on infinite scrolling lists.

I've updated the code to reflect both use cases